themes/docs-new/layouts/shortcodes/unified_mode.md

```diff
@@ -0,0 +1,192 @@
+
+### `:immediate` Notifications to Later Resources
+
+Unified mode delays immediate notifications to later resources.
+In unified mode the Chef Infra Client saves immediate notifications and executes them when the later resource is parsed. Immediate notifications to prior resources and delayed notifications behave the same as they did before unified mode.
+
+Chaining immediate notifications results in the same behavior as before unified mode. Changing the order of resources results does not change the results of a notification chain. Chained notifications in the same order during the Chef Infra Client run and immediately sends a notification to a target resource that was parsed earlier and saves a notification and sends it immediately after it parses a later resource.
+
+The changes to sending immediate notification could result in subtle changes to behaviors in some resources, but it not a breaking change to common patterns of writing resources.
+
+Chaining immediate notifications to later resources:
+
+```ruby
+remote_file "#{Chef::Config[:file_cache_path]}/myservice.tgz" do
+  source "http://acme.com/myservice.tgz"
+  notifies :extract, "archive_file[myservice.tgz]", :immediately
+end
+
+archive_file "#{Chef::Config[:file_cache_path]}/myservice.tgz" do
+  destination '/srv/myservice'
+  notifies :start, "service[myservice]", :immediately
+  action :nothing
+end
+
+service "myservice" do
+  action :nothing
+end
+```
+
+### `:before` Notifications to Later Resources
+
+Unified mode breaks resources with `before` notifications to later resources. In unified mode, you must declare a resource before you can act on it.
+
+Resources that subscribe to a `before` notification to a later resource must be declared after the resource that triggers the notification.
+
+This resource declares `before` notification to a later resource and will no longer work:
+
+```ruby
+package "myservice" do
+  notifies :stop, "service[myservice]", :before
+  notifies :start, "service[myservice]", :immediately
+end
+
+service "myservice" do
+  action :nothing
+end
+```
+
+Instead, declare the resource and then declare actions. For example:
+
+```ruby
+service "myservice" do
+  action :nothing
+end
+
+package "myservice" do
+  notifies :stop, "service[myservice]", :before
+  notifies :start, "service[myservice]", :immediately
+end
+```
+
+### Out Of Order Execution
+
+Unified mode  break custom resources that rely on the out-of-order execution of compile-time statements. Move any affected compile-time statements to the location in the code where they are intended to execute.
+
+Out-of-order execution is rare. At Chef, none of our custom resources broke during our conversion custom resources to unified mode. Instead, we discovered a few cases in which custom resource code was intended run in order, but Chef Infra Client executed it out of order. In these cases, unified mode fixed errors instead of introducing bugs.
```

Comment:
```suggestion
Out-of-order execution is rare. Internally at Chef, none of our custom resources broke during our migration to unified mode. Instead, we discovered a few cases in which custom resource code was intended to run in order, but Chef Infra Client executed it out of order. In these cases, unified mode fixed errors instead of introducing bugs.
```